File: LeetCode/1603设计停车系统.cpp
typedef struct {
    int big, mid, small;
} ParkingSystem;


ParkingSystem* parkingSystemCreate(int big, int medium, int small) {
    ParkingSystem *obj = (ParkingSystem *)malloc(sizeof(ParkingSystem));
    obj->small = small;
    obj->big = big;
    obj->mid = medium;
    return obj;
}

bool parkingSystemAddCar(ParkingSystem* obj, int carType) {
    if (carType == 1) {
        if (obj->big > 0) {
            obj->big--;
            return true;
        }
    } else if (carType == 2) {
        if (obj->mid > 0) {
            obj->mid--;
            return true;
        }
    } else if (carType == 3) {
        if (obj->small > 0) {
            obj->small--;
            return true;
        }
    }
    return false;
}

void parkingSystemFree(ParkingSystem* obj) {
    free(obj);
    return ;
}

/**
 * Your ParkingSystem struct will be instantiated and called as such:
 * ParkingSystem* obj = parkingSystemCreate(big, medium, small);
 * bool param_1 = parkingSystemAddCar(obj, carType);
 
 * parkingSystemFree(obj);
*/

File: LeetCode/1603设计停车系统.go
//switch case
type ParkingSystem struct {
	big    int
	medium int
	small  int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	var park ParkingSystem
	park.big = big
	park.medium = medium
	park.small = small
	return park
}

func (this *ParkingSystem) AddCar(carType int) bool {
	switch carType {
	case 1:
		{
			if this.big > 0 {
				this.big -= 1
				return true
			} else {
				return false
			}
		}
	case 2:
		{
			if this.medium > 0 {
				this.medium -= 1
				return true
			} else {
				return false
			}
		}
	case 3:
		{
			if this.small > 0 {
				this.small -= 1
				return true
			} else {
				return false
			}
		}
	}
	return false
}

/*------------------------使用数组-----------------------*/
type ParkingSystem struct {
	arr [3]int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{
		arr: [3]int{big, medium, small},
	}
}

func (this *ParkingSystem) AddCar(carType int) bool {
	this.arr[carType-1]--
	return this.arr[carType-1] >= 0
}